### ISSUE TITLE
Single line staff sticking engraving issues

### ISSUE THREAD

**Issue by zacjansheski** [musescore/MuseScore#20887]
### Issue type

Engraving bug (incorrect score rendering)

### Bug description

I am unable to align the sticking letters at the bottom, adjusting alignment in properties does not get the desired result of the letters aligning at the bottom.  Also the upper case letters intersect with the slurs for the flams

<img width="844" alt="Screenshot 2024-01-08 at 2 23 39 PM" src="https://github.com/musescore/MuseScore/assets/69917893/90413905-2678-462e-82fb-e8bff87302bf">

### Steps to reproduce

1. Write the music above on a single line staff 
2. Add>Text>Sticking with a combination of R, L, r, l
3. Select similar in staff, align baseline in properties.
4. the "r" sticking floats above the baseline of the other letters

### MuseScore Version

OS: macOS 14.1, Arch.: x86_64, MuseScore version (64-bit): 4.3.0-240080302, revision: github-musescore-musescore-4cbc720

### Regression

No.

### Operating system

Mac OS 14

**Comment by MarcSabatella** [comment]
It's *possible* to get automatic alignment, but the default settings don't give it by default and should.  See for example the discussions in https://musescore.org/en/node/308840 and https://musescore.org/en/node/292088.  Unless something has changed recently, I'm pretty sure the capability is there and it's just a matter of updating some style defaults.

**Comment by zacjansheski** [comment]
Do you mean the "offset" style settings?

They don't seem to have any affect currently

**Comment by MarcSabatella** [comment]
Yes, that much doesn't work indeed - see [this earlier comment](https://musescore.org/en/node/308840#comment-1018050).  To elaborate:

You are looking at the "text style" offsets as opposed to the "element" positions.  That's an unfortunate distinction with deep historical roots, but it has to do with the desire to have independent defaults for elements placed above/below the staff, and the fact that the latter feature was grafted on long after text styles were a thing.  So, some element types (like staff text) have position above/below settings in their own dedicated area of Format / Styles, as well as a separate offset in the text style (that we hope no one notices because the results of messing with that are not well-defined 😉).

Since the sticking element type was added in a bit of a rush to support the original MDL extension, it never got a proper set of above/below offsets exposed in the style dialog.  Instead, you can only set them via the Properties panel.  So, to get proper alignment *currently*, you need to set a more appropriate offset using the "Save as default style for this score" option in the Properties panel:

![image](https://github.com/musescore/MuseScore/assets/1799936/78467e61-5df4-499b-ae86-3a1e6442a8bf)

As far as I can tell there really *is* a separate position above & position below that is honored internally.  If the sticking you have selected when using "Save as default style for this score" is below the staff, you're actually setting the "position below" style setting, but if it's above, you're setting the "position above" style setting.  I've tested this and it seems to work, and to survive save/reload.  Neither of these are connected to position above/below (for some element types, the text style offset *is* connected to either position above or position below).

So, long story short, what needs to happen is:

1) the default for the "sticking position below" should be increased to something more reasonable, like 3 sp
2) there should probably be a sticking section of Format / Style (separate from the text style) to let you set position above/below directly

As a kind of separate matter, there is no special automatic alignment for sticking like there is for lyrics, so if you have one note way below the staff for some reason and it displaces the sticking, the others won't align to it.  The way the automatic alignment code works (or worked last time I worked on it), adding sticking to the list of auto-aligned element types wouldn't be hard, either.  Just not sure it would really be necessary, or even desirable, once the default offset is fixed.

**Comment by MarcSabatella** [comment]
The collision with slurs is another matter but if the default position below should probably is chosen to be low enough there that's not an issue.  Or, that could be an argument for doing the auto-align after all, if were want the default to be a bit higher but push everything on the system down if there is even one slur.
